superset/dashboards/filters.py

```diff
@@ -57,9 +58,9 @@ def apply(self, query: Query, value: Any) -> Query:
         return query.filter(
             or_(
                 Dashboard.created_by_fk  # pylint: disable=comparison-with-callable
-                == g.user.get_user_id(),
```

Comment:
The [get_user_id](https://github.com/dpgaspar/Flask-AppBuilder/blob/090ddb4645992fb390da93906301c63b204db0df/flask_appbuilder/security/sqla/models.py#L141-L146) method doesn't return the ID associated with the `User` object but rather the `g.user.id`. In this case they're the same but it's very misleading.